### PR TITLE
[WIP] Basic first pass implementation of http cloudevents.transport using bindings

### PR DESCRIPTION
**Pull request by n3wscott**
Signed-off-by: Scott Nichols <snichols@vmware.com>

### CODE REVIEW

File: pkg/cloudevents/transport/httpb/transport.go [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+	return ctx, nil, nil
+}
+
+// SetReceiver implements Transport.SetReceiver
+func (t *Transport) SetReceiver(r transport.Receiver) {
+	t.consumer = r
+}
+
+// SetConverter implements Transport.SetConverter
+func (t *Transport) SetConverter(c transport.Converter) {
```
Comment (slinkydeveloper) [inline pkg/cloudevents/transport/httpb/transport.go]: The converter handling IMO should be implemented directly by `binding.BindingTransport`: https://github.com/cloudevents/sdk-go/issues/315

File: pkg/cloudevents/transport/httpb/transport.go [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+// TODO: this is not what we want to use long term. For PoC.
+// Response is an http transport response.
+type Response struct {
+	StatusCode int
+	Message
+}
+
+// Message is an http transport message.
+type Message struct {
```
Comment (slinkydeveloper) [inline pkg/cloudevents/transport/httpb/transport.go]: Why do we need that? For the response?

**Comment by slinkydeveloper** [comment]
Trace extension implementations both on sender and receiver side can be implemented using a transformer. I suppose the flow should be something like:

* `Sender` should put in the encoded message the span from the event, if any
* `Receiver` should put in the generated message the span info, if any

File: pkg/cloudevents/transport/httpb/transport.go (hunk 1 above)
Comment (n3wscott) [reply pkg/cloudevents/transport/httpb/transport.go]: This is not that kind of converter, it allows you to implement a try again in the client side for custom converters

File: pkg/cloudevents/transport/httpb/transport.go (hunk 2 above)
Comment (n3wscott) [reply pkg/cloudevents/transport/httpb/transport.go]: It is needed for the struct in Response but that section of the code is not used yet.

**Comment by n3wscott** [comment]
This is suppose to be still WIP

**Pull request closed without merging**